qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -0,0 +1,126 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Polynomial Tensor class"""
+
+from __future__ import annotations
+from typing import Dict, Mapping
+from numbers import Number
+import numpy as np
+from qiskit.quantum_info.operators.mixins import (
+    LinearMixin,
+    AdjointMixin,
+    TolerancesMixin,
+)
+
+
+class PolynomialTensor(LinearMixin, AdjointMixin, TolerancesMixin):
+    """Polynomial Tensor class"""
+
+    def __init__(self, data: Mapping[str, np.ndarray | Number]):
+        copy_dict: Dict[str, np.ndarray] = {}
+
+        shapes = set()
+        for key, value in data.items():
+            if isinstance(value, Number):
+                value = np.asarray(value)
+
+            if len(value.shape) != len(key):
+                raise ValueError(
+                    f"Data key {key} of length {len(key)} does not match "
+                    f"data value matrix of dimensions {value.shape}"
+                )
+
+            dims = set(value.shape)
+
+            if len(dims) > 1:
+                raise ValueError(
+                    f"For key {key}: dimensions of value matrix are not identical {value.shape}"
+                )
+
+            shapes.update(dims)
+            copy_dict[key] = value
```

Comment:
Again, also just having seen #731, should we also think of a `_copy: bool` keyword-argument for this initializer? I would think this would be useful, since it would avoid multiple copies during addition and multiplication.

You can take a look at the that PR to see how this is handled there :+1:

Reply:
You can actually disregard this suggestion. I realized after talking to Steve offline, that you are not copying the (potentially massive) matrices anywhere which is good as it is now :+1: